scripts/get-external-data.py

```diff
@@ -135,13 +134,95 @@ def replace(self, new_last_modified):
         self._conn.commit()
 
 
+class Downloader:
+    def __init__(self):
+        self.session = requests.Session()
+        self.session.headers.update({'User-Agent': 'get-external-data.py/osm-carto'})
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *args, **kwargs):
+        self.session.close()
+
+    def _download(self, url, headers=None):
+        if url.startswith('file://'):
+            with open(url[7:], 'rb') as fp:
+                return DownloadResult(status_code = 200, content = fp.read(),
+                                      last_modified = str(os.fstat(fp.fileno()).st_mtime))
```

Comment:
Why this over requests_file?

Reply:
To eventually support last_modified with local files, too, as requests-file does not return the modification date at all.

Note it is not there yet, but I may implement in an extra commit this weekend if you'd like to see it.

Reply:
Ah, that makes sense.  I'll give the rest of this a fuller review.